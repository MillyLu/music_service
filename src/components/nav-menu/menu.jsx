import { useContext } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';

import * as Styled from './styles'
import { ThemeContext } from '../../ThemeProvider';
import { deleteUser } from '../../store/state';
import { ThemeSwitcher } from '../nav-themeSwitcher/themeSwitcher';





 export function Menu() {
    const { theme} = useContext(ThemeContext);
    const navigate = useNavigate();
    const dispatch = useDispatch();
    
    const path = "/login";

    const handleLogout = (e)  => {
        e.preventDefault();
        dispatch(deleteUser())
        localStorage.removeItem("token")
        localStorage.removeItem('name')
        navigate( path)
    }
    return(
        <Styled.NavMenu>
            <Styled.MenuList>
                <Styled.MenuItem>
                    <Styled.NavLink to='/' style={{ color: theme === "light" ? "#000000" : "#FFFFFF" }}>Главная</Styled.NavLink>
                </Styled.MenuItem>
                <Styled.MenuItem>
                    <Styled.NavLink to="/favorites" style={{ color: theme === "light" ? "#000000" : "#FFFFFF" }}>Мой треки</Styled.NavLink>
                </Styled.MenuItem>
                <Styled.MenuItem>
                    <Styled.NavLink onClick={handleLogout} style={{ color: theme === "light" ? "#000000" : "#FFFFFF" }}>Выйти</Styled.NavLink>
                </Styled.MenuItem>
                <Styled.MenuItem>
                    <ThemeSwitcher />
                </Styled.MenuItem>
            </Styled.MenuList>
        </Styled.NavMenu>
    )
}