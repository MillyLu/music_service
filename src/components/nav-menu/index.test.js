/* eslint-disable no-undef */
import { Provider } from 'react-redux';

import * as ShallowRenderer from 'react-shallow-renderer';
import { Router } from 'react-router-dom';
import { Menu } from './menu';


describe("<Menu />", () => {
    it("should render successfully and match snapshot", () => {

        const mockedUsedNavigate = jest.fn();

        jest.mock('react-router-dom', () => ({
           ...jest.requireActual('react-router-dom') ,
          useNavigate: () => mockedUsedNavigate,
        }));
        
        const store = {
            dispatch: jest.fn(),
            getState: jest.fn(() => ({})),
         };

         const shallowRenderer = ShallowRenderer.createRenderer();
         const renderer = shallowRenderer;
         renderer.render(<Provider store={store}><Router><Menu /></Router></Provider>);
         const result = renderer.getRenderOutput();
         expect(result).toMatchSnapshot();  


    });
  });



















