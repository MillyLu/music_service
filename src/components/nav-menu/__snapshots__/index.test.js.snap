// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render successfully and match snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "getServerState": undefined,
      "store": Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": Object {},
            },
          ],
        },
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Router>
    <Menu />
  </Router>
</Context.Provider>
`;
